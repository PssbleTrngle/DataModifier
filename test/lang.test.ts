import createTestAcceptor from './mock/TestAcceptor.js'
import setupLoader from './shared/loaderSetup.js'

const { loader } = setupLoader({ include: 'assets/*/lang/*.json' })

describe('replacing translation entries', () => {
   it('replaces using string value', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.replaceValue('diorite', 'bird poop', { lang: 'en_us' })

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('assets/minecraft/lang/en_us.json')).toMatchSnapshot('replaced diorite values')
   })

   it('replaces requires the case to match', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.replaceValue('Stone', 'rock', { lang: 'en_us', matchCase: true })

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('assets/minecraft/lang/en_us.json')).toMatchSnapshot('replaced matched case values')
   })

   it('emits custom values', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.addCustom('en_us', 'something.else', 'The Value')
      loader.lang.entryName('minecraft:item', 'minecraft:diamond', 'Sapphire')

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('assets/minecraft/lang/en_us.json')).toMatchSnapshot('custom values')
   })

   it('emits custom values together with replaced values', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.addCustom('en_us', 'something.else', 'The Value')
      loader.lang.entryName('minecraft:item', 'minecraft:diamond', 'Sapphire')
      loader.lang.replaceValue('Diamond', 'ruby')

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('assets/minecraft/lang/en_us.json')).toMatchSnapshot('custom values')
   })

   it('only modifies values of given mods', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.replaceValue('Iron', 'Steel', { lang: 'en_us', mod: ['create', 'farmersdelight'] })

      await loader.emit(acceptor)

      expect(acceptor.paths()).toHaveLength(2)
   })

   it('respects keepCase option', async () => {
      const acceptor = createTestAcceptor()

      loader.lang.replaceValue('Dark Oak', 'Mahagony', { lang: 'en_us', mod: ['minecraft'], keepCase: false })

      await loader.emit(acceptor)

      expect(acceptor.jsonAt('assets/minecraft/lang/en_us.json')).toMatchSnapshot('kept case values')
   })
})
