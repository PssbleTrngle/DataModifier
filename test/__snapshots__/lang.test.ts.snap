// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`replacing translation entries > emits custom values > custom values 1`] = `
{
  "item.minecraft.diamond": "Sapphire",
  "something.else": "The Value",
}
`;

exports[`replacing translation entries > emits custom values together with replaced values > custom values 1`] = `
{
  "advancements.story.mine_diamond.description": "Acquire rubys",
  "advancements.story.mine_diamond.title": "Rubys!",
  "advancements.story.shiny_gear.description": "Ruby armor saves lives",
  "advancements.story.shiny_gear.title": "Cover Me with Rubys",
  "block.minecraft.deepslate_diamond_ore": "Deepslate Ruby Ore",
  "block.minecraft.diamond_block": "Block of Ruby",
  "block.minecraft.diamond_ore": "Ruby Ore",
  "item.minecraft.diamond": "Sapphire",
  "item.minecraft.diamond_axe": "Ruby Axe",
  "item.minecraft.diamond_boots": "Ruby Boots",
  "item.minecraft.diamond_chestplate": "Ruby Chestplate",
  "item.minecraft.diamond_helmet": "Ruby Helmet",
  "item.minecraft.diamond_hoe": "Ruby Hoe",
  "item.minecraft.diamond_horse_armor": "Ruby Horse Armor",
  "item.minecraft.diamond_leggings": "Ruby Leggings",
  "item.minecraft.diamond_pickaxe": "Ruby Pickaxe",
  "item.minecraft.diamond_shovel": "Ruby Shovel",
  "item.minecraft.diamond_sword": "Ruby Sword",
  "something.else": "The Value",
  "subtitles.item.armor.equip_diamond": "Ruby armor clangs",
}
`;

exports[`replacing translation entries > replaces requires the case to match > replaced matched case values 1`] = `
{
  "advancements.story.mine_stone.title": "Rock Age",
  "block.minecraft.chiseled_stone_bricks": "Chiseled Rock Bricks",
  "block.minecraft.cracked_stone_bricks": "Cracked Rock Bricks",
  "block.minecraft.end_stone": "End Rock",
  "block.minecraft.end_stone_brick_slab": "End Rock Brick Slab",
  "block.minecraft.end_stone_brick_stairs": "End Rock Brick Stairs",
  "block.minecraft.end_stone_brick_wall": "End Rock Brick Wall",
  "block.minecraft.end_stone_bricks": "End Rock Bricks",
  "block.minecraft.infested_chiseled_stone_bricks": "Infested Chiseled Rock Bricks",
  "block.minecraft.infested_cracked_stone_bricks": "Infested Cracked Rock Bricks",
  "block.minecraft.infested_mossy_stone_bricks": "Infested Mossy Rock Bricks",
  "block.minecraft.infested_stone": "Infested Rock",
  "block.minecraft.infested_stone_bricks": "Infested Rock Bricks",
  "block.minecraft.mossy_stone_brick_slab": "Mossy Rock Brick Slab",
  "block.minecraft.mossy_stone_brick_stairs": "Mossy Rock Brick Stairs",
  "block.minecraft.mossy_stone_brick_wall": "Mossy Rock Brick Wall",
  "block.minecraft.mossy_stone_bricks": "Mossy Rock Bricks",
  "block.minecraft.smooth_stone": "Smooth Rock",
  "block.minecraft.smooth_stone_slab": "Smooth Rock Slab",
  "block.minecraft.stone": "Rock",
  "block.minecraft.stone_brick_slab": "Rock Brick Slab",
  "block.minecraft.stone_brick_stairs": "Rock Brick Stairs",
  "block.minecraft.stone_brick_wall": "Rock Brick Wall",
  "block.minecraft.stone_bricks": "Rock Bricks",
  "block.minecraft.stone_button": "Rock Button",
  "block.minecraft.stone_pressure_plate": "Rock Pressure Plate",
  "block.minecraft.stone_slab": "Rock Slab",
  "block.minecraft.stone_stairs": "Rock Stairs",
  "block.minecraft.stonecutter": "Rockcutter",
  "container.stonecutter": "Rockcutter",
  "item.minecraft.stone_axe": "Rock Axe",
  "item.minecraft.stone_hoe": "Rock Hoe",
  "item.minecraft.stone_pickaxe": "Rock Pickaxe",
  "item.minecraft.stone_shovel": "Rock Shovel",
  "item.minecraft.stone_sword": "Rock Sword",
  "stat.minecraft.interact_with_stonecutter": "Interactions with Rockcutter",
  "subtitles.ui.stonecutter.take_result": "Rockcutter used",
}
`;

exports[`replacing translation entries > replaces using string value > replaced diorite values 1`] = `
{
  "block.minecraft.diorite": "Bird poop",
  "block.minecraft.diorite_slab": "Bird poop Slab",
  "block.minecraft.diorite_stairs": "Bird poop Stairs",
  "block.minecraft.diorite_wall": "Bird poop Wall",
  "block.minecraft.polished_diorite": "Polished Bird poop",
  "block.minecraft.polished_diorite_slab": "Polished Bird poop Slab",
  "block.minecraft.polished_diorite_stairs": "Polished Bird poop Stairs",
}
`;

exports[`replacing translation entries > respects keepCase option > kept case values 1`] = `
{
  "block.minecraft.dark_oak_button": "Mahagony Button",
  "block.minecraft.dark_oak_door": "Mahagony Door",
  "block.minecraft.dark_oak_fence": "Mahagony Fence",
  "block.minecraft.dark_oak_fence_gate": "Mahagony Fence Gate",
  "block.minecraft.dark_oak_leaves": "Mahagony Leaves",
  "block.minecraft.dark_oak_log": "Mahagony Log",
  "block.minecraft.dark_oak_planks": "Mahagony Planks",
  "block.minecraft.dark_oak_pressure_plate": "Mahagony Pressure Plate",
  "block.minecraft.dark_oak_sapling": "Mahagony Sapling",
  "block.minecraft.dark_oak_sign": "Mahagony Sign",
  "block.minecraft.dark_oak_slab": "Mahagony Slab",
  "block.minecraft.dark_oak_stairs": "Mahagony Stairs",
  "block.minecraft.dark_oak_trapdoor": "Mahagony Trapdoor",
  "block.minecraft.dark_oak_wall_sign": "Mahagony Wall Sign",
  "block.minecraft.dark_oak_wood": "Mahagony Wood",
  "block.minecraft.potted_dark_oak_sapling": "Potted Mahagony Sapling",
  "block.minecraft.stripped_dark_oak_log": "Stripped Mahagony Log",
  "block.minecraft.stripped_dark_oak_wood": "Stripped Mahagony Wood",
  "item.minecraft.dark_oak_boat": "Mahagony Boat",
}
`;
